qualtran/bloqs/arithmetic.py

```diff
@@ -539,43 +539,83 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
 
 
 @frozen
-class OutOfPlaceAdder(Bloq):
+class OutOfPlaceAdder(GateWithRegisters, cirq.ArithmeticGate):
```

Comment:
Question: Is there an advantage to keeping some of the arithmetic bloqs as cirq.ArithmeticGates?

Reply:
Yes, they help with faster state vector cirq style simulations because `ArithmeticGate` has a `_apply_unitary_` overloaded.